### PR TITLE
Documentation for alert-related resources in LS terraform provider

### PR DESCRIPTION
**Pull request by ltyson**
What does this PR do?
-----------------------
Adds descriptions for a bunch of fields in the Lightstep terraform provider that currently have [zero documentation](https://registry.terraform.io/providers/lightstep/lightstep/latest/docs/resources/alert), which makes the provider really hard to use, and makes potentially useful features hard to discover.

Specifically I focused on the alert-related resources (the deprecated `lightstep_metric_alert` and the new `lightstep_alert`)

There should be no functional changes.

Why are you making this change?
-------------------------------
I was already in the area working on something else and decided to put some time into writing docs before bumping the provider version for the next release.

### CODE REVIEW

File: docs/resources/alert.md [hunk 1]
```diff
@@ -51,18 +51,18 @@ EOT
 
 ### Required
 
-- `name` (String)
-- `project_name` (String)
+- `name` (String) Title for the alert
+- `project_name` (String) Name of the project in which to create this alert
```
Comment (mattcarmody) [inline docs/resources/alert.md:55]: nit: would it be fair to say `Title of the alert`? ffti my compulsion for symmetry

File: docs/resources/alert.md [hunk 2]
```diff
@@ -74,9 +74,9 @@ EOT
 
 Optional:
 
-- `exclude_filters` (List of Map of String)
+- `exclude_filters` (List of Map of String) ID of the corresponding destination resource
 - `filters` (List of Map of String) Non-equality filters (operand: contains, regexp, etc)
-- `include_filters` (List of Map of String)
+- `include_filters` (List of Map of String) For alert queries that produce multiple group_by values, if at least one entry is specified for this field, the destination will only receive notification for group_by results that include the set of attributes specified here.
```
Comment (mattcarmody) [inline docs/resources/alert.md:79]: typo? I don't think `exclude_filters` is an ID

Comment (mattcarmody) [reply docs/resources/alert.md:79]: while we're here should we remove `exclude_filters` and `filters`? we only support equality

File: docs/resources/alert.md [hunk 3]
```diff
@@ -89,37 +89,37 @@ Read-Only:
...
 
 - `expression` (Block List, Min: 1, Max: 1) (see [below for nested schema](#nestedblock--composite_alert--alert--expression))
-- `name` (String)
+- `name` (String) Identifier for this sub-alert. Must be a single uppercase letter  (examples: A, B, C)
 - `query` (Block List, Min: 1, Max: 1) (see [below for nested schema](#nestedblock--composite_alert--alert--query))
 
 Optional:
 
-- `title` (String)
+- `title` (String) Optional free-form title for the alert
```
Comment (mattcarmody) [inline docs/resources/alert.md:105]: nit: should this be `sub-alert` for consistency with `name` field above?

File: docs/resources/metric_condition.md [hunk 4]
```diff
@@ -97,17 +97,17 @@ resource "lightstep_metric_condition" "beemo-requests" {
...
 ### Required
 
-- `expression` (Block List, Min: 1, Max: 1) (see [below for nested schema](#nestedblock--expression))
-- `metric_query` (Block List, Min: 1) (see [below for nested schema](#nestedblock--metric_query))
-- `name` (String)
-- `project_name` (String)
+- `expression` (Block List, Min: 1, Max: 1) Describes the conditions that should trigger the alert. (see [below for nested schema](#nestedblock--expression))
+- `metric_query` (Block List, Min: 1) Defines the alert query (see [below for nested schema](#nestedblock--metric_query))
+- `name` (String) Title for the alert
+- `project_name` (String) Name of the project in which to create this alert
```
Comment (mattcarmody) [inline docs/resources/metric_condition.md:103]: nit: same as above comment wrt `Title for` vs `Name of` immediately below

File: lightstep/resource_alert_test.go [hunk 5]
```diff
@@ -578,11 +559,6 @@ EOT
       value = "catlab"
     }]
 
-	filters = [{
-		  key   = "service_name"
-		  value = "frontend"
-		  operand = "contains"
-	}]
```
Comment (mattcarmody) [inline lightstep/resource_alert_test.go:585]: thank you!

**Review by mattcarmody — APPROVED** [review]
LGTM in general, left a few minor comments

File: docs/resources/alert.md (hunk 2 above)
Comment (ltyson) [reply docs/resources/alert.md:79]: heh, yeah, this was a mangled diff from trying to remove `exclude_filters` and `filters` and then thinking better of it and removing those changes -- thanks for noticing! and yeah, I'll remove the filters in a separate PR to avoid holding up this one.

File: client/stream_alerting_rule.go [hunk 6]
```diff
@@ -58,7 +58,7 @@ type StreamAlertingRuleResponseRelationships struct {
 func (c *Client) CreateAlertingRule(
 	ctx context.Context,
 	projectName string,
-	updateInterval int,
+	updateIntervalMs int,
```
Comment (ltyson) [inline client/stream_alerting_rule.go:61]: this is unrelated, it was just bugging me that there were no units on this thing

**Comment by rkuchan** [comment]
@ltyson, I just pushed some commits -- I did a tech-writery pass, adding periods everywhere, linking to the main docs where possible, and using present tense over future tense. 

There's one spot about `hidden_queries` where I'm not confident in a change. I'll leave a comment on it.

**Review by rkuchan — APPROVED** [review]

File: docs/resources/alert.md [hunk 7]
```diff
@@ -135,8 +136,8 @@ Required:
...
-- `hidden_queries` (Map of String) An optional map of sub-query names in the query_string to a boolean string to hide/show that query. If specified, the map must have an entry for all named sub-queries in the query_string. A value of "true" indicates the query should be hidden. Example: `hidden_queries = {  "a" = "true",  "b" = "false" }`.
+- `display_type_options` (Block Set, Max: 1) Applicable options vary depending on the display type. Please see the Lightstep documentation for a full description. (See [below for nested schema](#nestedblock--composite_alert--alert--query--display_type_options).)
+- `hidden_queries` (Map of String) An optional map of sub alert names in the query_string to a boolean string to hide/show that query. If specified, the map must have an entry for all named sub alerts in the query_string. A value of "true" indicates the query should be hidden. Example: `hidden_queries = {  "a" = "true",  "b" = "false" }`.
```
Comment (rkuchan) [inline docs/resources/alert.md:140]: Three things:

- In the main docs, we decided to use sub alert (not sub-alert). I went ahead and changed things here for consistency with the main docs. I'm happy to revisit that hyphen decision whenever though.
- For consistency with the identifier for sub alerts mentioned elsewhere, I changed this from sub query to sub alert. Is that accurate? I will change it back if not. 
- Should the example here use upper case letters?

**Comment by ltyson** [comment]
ok, it's not 100% complete but it's a net improvement

File: docs/resources/alert.md (hunk 7 above)
Comment (ltyson) [reply docs/resources/alert.md]: (sub-queries are a different thing so I left those as-is)
